pkg/descheduler/descheduler.go

```diff
@@ -60,7 +60,7 @@ func Run(rs *options.DeschedulerServer) error {
 	return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
 }
 
-type strategyFunction func(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, evictLocalStoragePods bool, podEvictor *evictions.PodEvictor)
+type strategyFunction func(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, opts strategies.Options, podEvictor *evictions.PodEvictor)
```

Comment:
Actually, there's nothing forbidding to move `strategyFunction` and `strategyFuncs` under `"sigs.k8s.io/descheduler/pkg/descheduler/strategies"`. @damemi WDYT?

Reply:
If we're ok with that, I would also suggest pushing down the options into its own `sigs.k8s.io/descheduler/pkg/descheduler/strategies/options` package, as we should then also be able to use this from the pod package and mitigate the circular import problem. This would mean that we could then pass in the opts directly to `ListEvictablePodsOnNode`, instead of passing in a flag at a time, as it does now.

Reply:
I've now added a commit that does the Options push-down and cleans up the `ListEvictablePodOnNode` usage. While I'm not thrilled about adding another package, I think this is going in the right direction.